sklearn/neural_network/multilayer_perceptron.py

```diff
@@ -97,6 +97,10 @@ def _forward_pass(self, activations):
         activations : list, length = n_layers - 1
             The ith element of the list holds the values of the ith layer.
         """
+        # local import to allow custom activations set via this
+        # private variable in _base
+        from ._base import ACTIVATIONS
```

Comment:
First to remove `ACTIVATIONS` and `DERIVATIVES` from a public import path. Second to allow monkeypatching it. If you import it at the top level, it's imported together with `MLPClassifier`, and then changing `neural_network._base.ACTIVATION` no longer have an effect on `MLPClassifier`.

Reply:
You are right. Actually the above example will work even with top level import. The problem is pytest monkeypatching, if I don't do local imports tests using it will fail, since I monkeypatch `ACTIVATIONS` instead of just updating it to be sure not to affect the global state or other tests.

I removed the comment as it was not accurate, but I guess in any case it doesn't hurt not to expose those variables at top level?

Reply:
> It's not great, but having a local import so that we can write a test in a certain way sounds like a strange design to me

Fair enough, I can also just revert https://github.com/scikit-learn/scikit-learn/pull/14815/commits/7519f1412eeb6ab99d5ba0dfdafc1f893642ca9a and we will be back where we started but with an example in the issue using the private API :)